### PR TITLE
Adding Nextalign in Nextclade suite

### PR DESCRIPTION
**Pull request by raguertin96**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/nextclade/nextalign.xml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    --nuc-mismatches-allowed '${nuc_mismatches_allowed}'
+    --aa-seed-length '${aa_seed_length}'
+    --aa-min-seeds '${aa_min_seeds}'
+    --aa-seed-spacing '${aa_seed_spacing}'
+    --aa-mismatches-allowed '${aa_mismatches_allowed}'
+    ]]></command>
+    <inputs>
+        <expand macro="reference"/>
+        <param argument="--sequences" type="data" format="fasta" label="FASTA file with input sequences"/>
+        <param name="output_insertions" type="boolean" checked="true" label="Output insertion sequences?" help="Outputs stripped insertions relative to reference as CSV"/>
```
Comment (bgruening) [inline tools/nextclade/nextalign.xml:43]: you could also add argument="--output-insertions" here

File: tools/nextclade/nextalign.xml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            <param name="output_insertions" value="true"/>
+            <param name="sequences" value="subsampled.fasta"/>
+            <conditional name="translation">
+                <param name="translation_select" value="yes"/>
+                <param name="genes" value="E,M,N,ORF10,ORF14,ORF1a,ORF1b,ORF3a,ORF6,ORF7a,ORF7b,ORF8,ORF9b,S"/>
+                <param name="genemap" value="genemap.gtf" ftype="gtf"/>
+            </conditional>
+            <output name="output_fasta" file="output.fasta" sort="true"/>
+            <output name="output_csv" file="insertions.csv" sort="true"/>
+            <output_collection name="translations" type="list">
```
Comment (bgruening) [inline tools/nextclade/nextalign.xml:137]: you could add here also the number of elements of the collection as test

File: tools/nextclade/nextalign.xml [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            <param name="output_insertions" value="false"/>
+            <param name="sequences" value="subsampled.fasta"/>
+            <conditional name="translation">
+                <param name="translation_select" value="no"/>
+            </conditional>
+            <output name="output_fasta" file="output.fasta" sort="true"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+Nextalign is a viral genome sequence alignment algorithm used in Nextclade.
```
Comment (bgruening) [inline tools/nextclade/nextalign.xml:186]: That could be improved a little bit ... That might help: rst.ninjs.org

**Review by bgruening — COMMENTED** [review]
@raguertin96 that is a pretty damn good tool! Very well done! :)

I just have some tiny comments.

Thanks a lot!

File: tools/nextclade/macros.xml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        <conditional name="reference_source">
+            <param name="reference_source_selector" type="select" label="Choose the source for the reference genome">
+                <option value="cached">Use a built-in genome</option>
+                <option value="history">Use a genome from history</option>
+            </param>
+            <when value="cached">
+                <param name="ref_file" type="select" label="Using reference genome" help="Select genome from the list">
+                    <options from_data_table="all_fasta">
+                        <filter type="sort_by" column="2"/>
+                        <validator type="no_options" message="No reference genomes are available"/>
```
Comment (bernt-matthias) [inline tools/nextclade/macros.xml:46]: The validator in `param` should be sufficient.

File: tools/nextclade/nextalign.xml [hunk 5]
```diff
@@ -0,0 +1,193 @@
+<tool id="nextalign" name="NextAlign" version="@NEXTALIGN_VERSION@+galaxy0" python_template_version="3.5">
```
Comment (bernt-matthias) [inline tools/nextclade/nextalign.xml:1]: Can you add a profile >= 20.01?

File: tools/nextclade/nextalign.xml [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    --nuc-seed-spacing '${nuc_seed_spacing}'
+    --nuc-mismatches-allowed '${nuc_mismatches_allowed}'
+    --aa-seed-length '${aa_seed_length}'
+    --aa-min-seeds '${aa_min_seeds}'
+    --aa-seed-spacing '${aa_seed_spacing}'
+    --aa-mismatches-allowed '${aa_mismatches_allowed}'
+    ]]></command>
+    <inputs>
+        <expand macro="reference"/>
+        <param argument="--sequences" type="data" format="fasta" label="FASTA file with input sequences"/>
```
Comment (bgruening) [inline tools/nextclade/nextalign.xml:42]: @raguertin96 do you know if this works on fasta.gz files?

[the remark this answers is not in the document]
Comment (NatPRoach) [reply tools/nextclade/nextalign.xml:42]: Not sure Bjoern, @raguertin96 and I can test that and get back to you shortly though.

[the remark this answers is not in the document]
Comment (NatPRoach) [reply tools/nextclade/nextalign.xml:42]: Seems to not accept fasta.gz @bgruening

**Review by bgruening — APPROVED** [review]

File: tools/nextclade/macros.xml [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<macros>
+    <token name="@NEXTALIGN_VERSION@">0.1.6</token>
```
Comment (bernt-matthias) [inline tools/nextclade/macros.xml:2]: Can you change this to `@TOOL_VERSION@` as described in https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#tool-versions

File: tools/nextclade/nextalign.xml [hunk 8]
```diff
@@ -0,0 +1,203 @@
+<tool id="nextalign" name="NextAlign" version="@NEXTALIGN_VERSION@+galaxy0" python_template_version="3.5" profile="20.01">
```
Comment (bernt-matthias) [inline tools/nextclade/nextalign.xml:1]:
```suggestion
<tool id="nextalign" name="NextAlign" version="@NEXTALIGN_VERSION@+galaxy0" profile="20.01">
```

File: tools/nextclade/.shed.yml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  name: "suite_nextclade"
+  description: Identify differences between your sequences and a reference sequence used by Nextstrain
+long_description: |
+  Nextclade is a tool that identifies differences between your sequences
+  and a reference sequence used by Nextstrain, uses these differences to
+  assign your sequences to clades, and reports potential sequence quality
+  issues in your data. You can use the tool to analyze sequences before
+  you upload them to a database, or if you want to assign Nextstrain
+  clades to a set of sequences.
+
```
Comment (bernt-matthias) [inline tools/nextclade/.shed.yml:30]:
```suggestion
```

duplicated

**Comment by bgruening** [comment]
@raguertin96 can you please address those last review comments? Thanks!

File: tools/nextclade/macros.xml (hunk 7 above)
Comment (raguertin96) [reply tools/nextclade/macros.xml]: Done in https://github.com/galaxyproject/tools-iuc/pull/3561/commits/68f393035c3a4e42de15020be964fa1438fce896

Sorry for the delay, let me know if there is anything else.

**Comment by bgruening** [comment]
Very cool. Thanks a lot!

**Pull request merged**